### ISSUE TITLE
dns configuration option does not work on docker-compose==1.6.0

### ISSUE THREAD

**Issue by mmarzantowicz** [docker/compose#2847]
Step 1:
```
$ cat docker-compose.yml 
version: "2"

services:
    testnet:
        image: fedora:23
        dns: 192.168.1.1
        command: /bin/bash
```
Step 2:
```
$ docker-compose -f docker-compose.yml run testnet
```
Step 3:
```
[root@bbd9f61098fb /]# cat /etc/resolv.conf 
search xxx.yyy.tld
nameserver 127.0.0.11
options ndots:0
```
As you can see, `dns` option is ignored by `docker-compose` and some garbage is set instead. This prevents my containers from starting up because they require to use mentioned DNS server. BTW, what kind of shit is 127.0.0.11, and how it get to resolve.conf?

With version 1 of yaml file everything is working as expected (nameserver is set correctly).

**Comment by dnephin** [comment]
This seems to be an issue with docker/docker (https://github.com/docker/docker/issues/20101), but we need more information to be able to triage it. It doesn't reproduce for me either.

If you could provide more information to https://github.com/docker/docker/issues/20101 about your environment (Host OS, `docker info`, `cat /etc/resolv.conf`, etc) that would help get the issue resolved.

`127.0.0.11` is a loopback IP address. It is used by the embedded docker DNS server.

**Comment by mavenugo** [comment]
@mmarzantowicz also pls confirm if you are using fedora/centos/RHEL ?
If yes, we have a known issue and the workaround is to disable `firewalld`.

**Comment by mavenugo** [comment]
@mmarzantowicz btw, if you can read about the embedded-DNS server (since you seem to be interested in knowing about 127.0.0.11 :) ) : 
https://github.com/docker/docker/blob/ef52a53c34d9f128d9435d031442509837c25c02/docs/userguide/networking/dockernetworks.md#docker-embedded-dns-server

**Comment by mmarzantowicz** [comment]
@mavenugo , my host OS is Arch Linux and only guests are on Fedora or anything RH-based. I do not have any firewalls enabled so docker rules. I now understand how it works but it looks like I have different connectivity issue to solve that I primarily thought. Thanks for your comments.

**Comment by mmarzantowicz** [comment]
I'm closing this issue as I'm no longer able to reproduce this bug. I'm not sure why some apps in container were complaining about DNS server at 127.0.0.11. I'll reopen this issue in case I find it possible to reproduce mentioned failure again.

Thanks for all your support.

**Comment by mbelang** [comment]
I have the exact same problem.
When starting the container with docker run --dns it works perfectly.
When I set the dns in the docker-compose the resolv.conf on the container is not updated...

docker compose file

```
version: '2'
services:
  app:
    image: toto
    container_name: toto
    restart: always

    hostname: toto.example.org
    dns:
      - 192.168.1.1
    ports:
      - "8000:80"

    volumes:
      - "/etctoto:/etc/toto"

    logging:
      driver: syslog
      options:
        syslog-tag: "{{.Name}}.{{.ID}}"
```

docker-compose v1.6
docker info :
```
Containers: 2
 Running: 2
 Paused: 0
 Stopped: 0
Images: 4
Server Version: 1.10.2
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 66
 Dirperm1 Supported: false
Execution Driver: native-0.2
Logging Driver: json-file
Plugins: 
 Volume: local
 Network: bridge null host
Kernel Version: 3.13.0-24-generic
Operating System: Ubuntu 14.04 LTS
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 7.798 GiB
Name: node02
ID: XNGO:L6KC:FGXE:T34M:2K4H:MTTL:WL6V:VOAQ:THV7:YFCG:3ZGP:N5WA
WARNING: No swap limit support

```

**Comment by ernetas** [comment]
Please, reopen this. I also can confirm this issue on Ubuntu 14.04. Restarting Docker service is what solves the issue.

**Comment by mmarzantowicz** [comment]
@ernetas, please check your issue is not related to one mentioned by @dnephin in response to my original post. I'm not aware of what version of `docker`/`docker-compose` are you having issues with on Ubuntu but it already might have been fixed upstream.

**Comment by mbelang** [comment]
I investigate on my side and I found my problem. I cannot speak for everybody but here is what I found:

Setting the property: net: "host" in the docker-compose file or --net host in the docker run command will result in the ports not displayed when doing a docker ps. It is normal because setting this property let the container inherits the host network settings.

As per documentation, this configuration is not recommend so I change it to: net: "bridge" (default if not specified) and added a DNS on my docker host to solve my communication problems.

**Comment by smiklosovic** [comment]
I am on CentOS 7 and I have this problem. I am on docker-compose 1.7.1 and Docker 1.11.1.

dns: 8.8.8.8 is not added into /etc/resolv.conf at all and I am not able to add any host to /etc/hosts as well by linking some other service. I have custom bridge network.

**Comment by smiklosovic** [comment]
I am not completely sure what is wrong but I am not able to add e.g. dns record to /etc/hosts at all even I am using just docker-compose up -d on a single docker-compose.yml with one service.

I am trying to setup nexus repo:

    version: '2' 
    services:
        nexus:
            hostname: nexus
            domainname: somedomain.net
            container_name: nexus
            image: sonatype/nexus:oss
            networks:
                server-net:
                    ipv4_address: 172.19.0.7
            ports:
                - "8081:8081"
            volumes:
                -  /srv/docker/nexus/nexus-data:/sonatype-work
            dns: 8.8.8.8
            dns_search:
                - somedomain.net
            environment:
                - MAX_HEAP=768m

    networks:
        server-net:
            driver: bridge
            ipam:
                driver: default
                config:
                    - subnet: 172.19.0.0/24
                      gateway: 172.19.0.1

**Comment by jamyouss** [comment]
+1 same issue for me

**Comment by grzegorz-gn** [comment]
How to slove dns problem when using docker-compose ?
When i changed dns in /etc/resolv.conf - every links to other containers not working.

**Comment by mbelang** [comment]
Install dnsmasq on the host and you should be alright and put the container to net: bridge.

**Comment by Coobers** [comment]
This issue is a critical one, please open it ASAP.
Without ability to properly define DNS, compose is a harming manner, especially on production. 
Please consider removing 1.6 until problem is not fixed.

**Comment by aanand** [comment]
DNS is configured differently when using a custom network (as is the case when using v2 of the Compose file format), but the `dns` setting is still respected. From the [Docker networking docs](https://docs.docker.com/engine/userguide/networking/configure-dns/):

> The IP addresses passed via the `--dns` option is used by the embedded DNS server to forward the DNS query if embedded DNS server is unable to resolve a name resolution request from the containers. These `--dns` IP addresses are managed by the embedded DNS server and will not be updated in the container's `/etc/resolv.conf` file.

**Comment by Coobers** [comment]
You can be right if DNS are down by docker's bug:
https://github.com/docker/docker/issues/11407 
But I'll wait till this one is resolved and then try again.

Till then docker, so long, and thanks for all the fish...

**Comment by bjackson** [comment]
Anyone found a fix for this?

**Comment by Coobers** [comment]
No fix, you need to change host dns network settings to get a result.

**Comment by jesuGMZ** [comment]
I had the same problem under docker-compose 1.11. Solved adding `network_mode` to the yaml configuration.

**Comment by umlen-B** [comment]
Hello!
I am facing the same issue in Ubuntu 14.04 under proxy.

> **$ docker version**
Client:
 Version:      17.03.1-ce
 API version:  1.27
 Go version:   go1.7.5
 Git commit:   c6d412e
 Built:        Fri Mar 24 00:40:33 2017
 OS/Arch:      linux/amd64
Server:
 Version:      17.03.1-ce
 API version:  1.27 (minimum version 1.12)
 Go version:   go1.7.5
 Git commit:   c6d412e
 Built:        Fri Mar 24 00:40:33 2017
 OS/Arch:      linux/amd64

> **$ docker-compose version**
docker-compose version 1.11.2, build dfed245
docker-py version: 2.2.1
CPython version: 2.7.6
OpenSSL version: OpenSSL 1.0.1f 6 Jan 2014

> **$ docker info**
Containers: 61
 Running: 0
 Paused: 0
 Stopped: 61
Images: 8
Server Version: 17.03.1-ce
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 130
 Dirperm1 Supported: false
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local
 Network: bridge host macvlan null overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 4ab9917febca54791c5f071a9d1f404867857fcc
runc version: 54296cf40ad8143b62dbcaa1d90e520a2136ddfe
init version: 949e6fa
Security Options:
 apparmor
Kernel Version: 3.13.0-77-generic
Operating System: Ubuntu 14.04.5 LTS
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 7.575 GiB
Name: HP
ID: PMD5:MD5B:XRP5:2KJ2:MVQW:XIQW:XFBJ:JZDP:TERL:4VMN:TX6F:EAWD
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Http Proxy: https://user:pass@172.31.102.29:3128/
Https Proxy: https://user:pass@172.31.102.29:3128/
Username: johnsonbhengra
Registry: https://index.docker.io/v1/
WARNING: No swap limit support
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
Experimental: false

> **$ uname -a**
Linux HP 3.13.0-77-generic #121-Ubuntu SMP Wed Jan 20 10:50:42 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux


I tried updating the dns in /etc/default/docker
I updated /etc/resolv.conf
Even tried commenting #dnsmasq
But no success, still the same error
![error](https://cloud.githubusercontent.com/assets/13407561/24668149/76c1b2da-1983-11e7-8bcc-54f774a3cb56.png)

#**Please Help!!**

**Comment by stefanocog** [comment]
Same problem, docker version 17.04.0-ce, docker-compose version 1.12.0, build b31ff33

**Comment by bush** [comment]
Same issue here:

Client:
 Version:      17.05.0-ce
 API version:  1.29
 Go version:   go1.7.5
 Git commit:   89658be
 Built:        Thu May  4 22:06:06 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.05.0-ce
 API version:  1.29 (minimum version 1.12)
 Go version:   go1.7.5
 Git commit:   89658be
 Built:        Thu May  4 22:06:06 2017
 OS/Arch:      linux/amd64
 Experimental: false

**Comment by arnaudblancher** [comment]
Hello,

i have the same  on Ubuntu 16.04.2 LTS
with docker-ce 17.03.1

best regards

**Comment by bodsch** [comment]
The same problem here
alpine 3.6
Docker version 17.03.1-ce
docker-compose version 1.13.0

**Comment by cyc115** [comment]
same problem on ubuntu 16.04 EC2 instance 
``` 
Client:
 Version:      17.05.0-ce
 API version:  1.29
 Go version:   go1.7.5
 Git commit:   89658be
 Built:        Thu May  4 22:10:54 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.05.0-ce
 API version:  1.29 (minimum version 1.12)
 Go version:   go1.7.5
 Git commit:   89658be
 Built:        Thu May  4 22:10:54 2017
 OS/Arch:      linux/amd64
 Experimental: false
```
why is this unsolved issue closed ?

**Comment by anlek** [comment]
@dnephin any way to reopen this issue? I'm having a similar problem :(

**Comment by lauriskuznecovs** [comment]
Adding this into **docker-compose.yml** solved my DNS issues:

```
dns: 8.8.8.8
dns_opt:
  - use-vc
  - no-tld-query
```

**Comment by jurrian** [comment]
Setting the dns in docker-compose.yml is one option.
The real problem is the /etc/resolv.conf file on the host system, this generated by a network manager and should be a symlink.
Docker containers use this file on the host system:

> In the absence of the --dns=IP_ADDRESS..., --dns-search=DOMAIN..., or --dns-opt=OPTION... options, Docker uses the /etc/resolv.conf of the host machine (where the docker daemon runs). While doing so the daemon filters out all localhost IP address nameserver entries from the host’s original file.
> 
> Filtering is necessary because all localhost addresses on the host are unreachable from the container’s network. After this filtering, if there are no more nameserver entries left in the container’s /etc/resolv.conf file, the daemon adds public Google DNS nameservers (8.8.8.8 and 8.8.4.4) to the container’s DNS configuration. If IPv6 is enabled on the daemon, the public IPv6 Google DNS nameservers will also be added (2001:4860:4860::8888 and 2001:4860:4860::8844).

In my case upgrading to Ubuntu 17.04 probably triggered NetworkManager to misconfigure /etc/resolv.conf. As is [shown here](https://askubuntu.com/questions/137037/networkmanager-not-populating-resolv-conf), `sudo dpkg-reconfigure resolvconf` fixed the issue for me.

**Comment by Stark-X** [comment]
As @jesuGMZ  said, I added `network_mode` to my docker-compose.yml solved the issue.
FYI.
```
docker-compose version 1.11.2, build dfed245
docker-py version: 2.1.0
CPython version: 2.7.13
OpenSSL version: OpenSSL 1.0.1t  3 May 2016
```
```
Network: null overlay bridge host
```
```
Client:
 Version:      1.12.3
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   6b644ec
 Built:
 OS/Arch:      linux/amd64

Server:
 Version:      1.12.3
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   6b644ec
 Built:
 OS/Arch:      linux/amd64
```

**Comment by jbpin** [comment]
I'll be interested by a solution to this issue...

**Comment by bieniek** [comment]
@jurrian 
Thanks it solves my problem.
Docker version 1.13.1, build 092cba3
docker-compose version 1.8.0, build unknown
Ubuntu 16.04.2 LTS

**Comment by devjb** [comment]
Second this,

dns resolving for internal addresses is not working:

```
version: "3"
services:
  client1:
    build: ./testclient
    cap_add:
     - NET_ADMIN
    dns: 10.100.128.77
    image: "oclient:testing"
    container_name: "client1"
    privileged: true
    tty: true
    networks:
      nwclient1: {}
    volumes:
     - ./../:/ovpnconfig
    command: openvpn /ovpnconfig/client1.ovpn
  client2:
    build: ./testclient
    cap_add:
     - NET_ADMIN
    dns: 10.100.128.77
    image: "oclient:testing"
    container_name: "client2"
    privileged: true
    tty: true
    networks:
      nwclient2: {}
    volumes:
     - ./../:/ovpnconfig
    command: openvpn /ovpnconfig/client2.ovpn
networks:
  nwclient1:
    #internal: true
  nwclient2:
    #internal: true
```
Container:
```
dc run --rm client1 bash
bash-4.4# cat /etc/resolv.conf
search de.company.int
nameserver 127.0.0.11
options ndots:0
bash-4.4#
```
Host:
```
cat /etc/resolv.conf
# Generated by NetworkManager
search de.company.int
nameserver 10.100.128.77
nameserver 10.100.128.80
```

Starting regular docker run with --dns option works.

**Comment by donbowman** [comment]
Does anyone have a workaround for this?
docker 18.05.0-ce-dev
ubuntu 18.04
docker-compose version 1.20.1, build 5d8c71b      

similar to above, when docker-compose creates the networks, it sets the resolv.conf to 127.0.0.11 in the containers, but they don't have anything listening to that inside the container, so it fails.

using docker run w/ default (default bridge, default dns) works.

**Comment by zeigerpuppy** [comment]
there is a a workaround using volume mapping to pull in a `resolv.conf`, e.g.: 

```
version: '2'
services:
    test.base:
        container_name: test.base
        image: docker.pnet.ch/r-base:latest
        restart: on-failure
        networks:
            - mynet
        volumes:
            - ./resolv.conf:/etc/resolv.conf
```

see see https://stackoverflow.com/questions/41243665/docker-compose-does-not-update-resolv-conf

This issue should really be fixed, but the workaround is effective

**Comment by Sytten** [comment]
Why is this closed, it is clearly not fixed...
Same problem on Centos 7 with docker-compose 1.21.2

**Comment by comzyh** [comment]
Still, have the same issue

```
Ubuntu 18.04 LTS
Docker version 18.03.1-ce, build 9ee9f40
docker-compose version 1.21.2, build a133471
```

```
docker run --rm -it --dns 1.1.1.1 alpine cat /etc/resolv.conf
```
it shows 

> nameserver 1.1.1.1


When using docker-compose
`tee docker-compose.yml`

```
version: '3'
services:
  test-dns:
    image: alpine
    command: cat /etc/resolv.conf
    dns: 1.1.1.1
```


it shows:

> Creating ss-compose_test-dns_1 ... done
> Attaching to ss-compose_test-dns_1
> test-dns_1  | nameserver 127.0.0.11
> test-dns_1  | options ndots:0
> ss-compose_test-dns_1 exited with code 0

**Comment by Sytten** [comment]
See the related posts, there is a PR opened to fix it, but not yet merged.

**Comment by jrichard8** [comment]
For me the problem is not fix.
When I with a 'docker run --dns' command, the DNS work but not with the docker-compose file.

**Comment by Sytten** [comment]
Just specify the DNS in your docker-compose file and it ill work, the PR is not yet merged for some reason. Feel free to comment on it so they finally merge it.

**Comment by Sytten** [comment]
This issue has been solved, the PR was merged. It should be release by now. Have you updated your docker?

**Comment by dpizzle** [comment]
Yes, I've updated the Centos host, see below. I tested the same compose file on a OSX host running `Docker version 18.09.0` and `docker-compose version 1.23.1` and don't have this issue.
```
Kernel Version: 3.10.0-862.11.6.el7.x86_64
Operating System: CentOS Linux 7 (Core)

Docker version 18.09.0, build 4d60db4
docker-compose version 1.23.1, build b02f1306
```

Host - /etc/resolv.conf

```
search acme.local acme.new.local
nameserver 10.1.1.1
nameserver 10.2.2.2
```


Container - /etc/resolv.conf

```
search acme.local acme.new.local
nameserver 127.0.0.11
options ndots:0
```

Test within container -

```
# ping example
ping: bad address 'example'

# ping example.acme.new.local
PING example.acme.new.local (10.1.2.3): 56 data bytes
64 bytes from 10.1.2.3: seq=0 ttl=59 time=3.541 ms
```

**Comment by dpizzle** [comment]
Resolved my issue by creating and adding the below to `/etc/docker/daemon.json` on the Centos:7 host.
```
{
  "dns-opts":[
    "ndots:1"
  ]
}
```

**Comment by mcblum** [comment]
@dpizzle were you using swarm? No matter what I do, I can't get any DNS resolution within my containers. Driving me crazy!

**Comment by dpizzle** [comment]
Hi @mcblum, I wasn’t using swarm, just standalone docker. The above snippet worked for me.

**Comment by mcblum** [comment]
Thank you!

**Comment by NicolasDorier** [comment]
this issue is not solved @mmarzantowicz open it again,

**Comment by uy-rrodriguez** [comment]
Dear all,

We were facing the same issue. Starting a container with `docker run --dns` was working fine, the file `/etc/resolv.conf` inherits the DNS servers of the host. However, using the `dns` option in `docker-compose.yml` leaves the default content:

```
search name.of.our.domain
nameserver 127.0.0.11
options ndots:0
```

After adding `network_mode: bridge`  in the YML file it works, the file `resolv.conf` gets the configured DNS servers. I thought `bridge` was the option used by default.

I wonder whether this could be documented somewhere alongside the `dns` option? I couldn't find the solution until reading all different workarounds here.

**Comment by dozerman94** [comment]
@uy-rrodriguez  I always thought the default network mode was bridge but turns out it's not. It is mentioned in the [compose documentation](https://docs.docker.com/compose/networking/):

> By default Compose sets up a single network for your app. Each container for a service joins the default network and is both reachable by other containers on that network, and discoverable by them at a hostname identical to the container name.

**Comment by uy-rrodriguez** [comment]
Thank you for the link @dozerman94 . When I was looking for help with the problem above, the point I couldn't find in the official docs was the need of selecting `network_mode: bridge` to have your custom `dns` applied into `/etc/resolv.conf`. Although I agree it makes sense not to touch that file when the services' network is private and only shared between them, it feels weird to change a Docker configuration option and not see any change on your image.

**Comment by nathanfranke** [comment]
This marks year five.

**Comment by exptom** [comment]
I'm getting this same problem with docker-compose. I've had to use the workaround to map a `resolv.conf` file into the container.

**Comment by jvies** [comment]
I can confirm that the binding or resolv.conf works as does the usage of bridged network.

**Comment by MohammedNoureldin** [comment]
@uy-rrodriguez I get that `network_mode` and `networks` cannot be combined. docker-compose v3.9

**Comment by NicolasDorier** [comment]
Can somebody reopen this? This is 5 years already, people still having problem with it.  @ulyssessouza

**Comment by MohammedNoureldin** [comment]
This definitely still exists, please reopen.

**Comment by NicolasDorier** [comment]
@ndeloof  said this will be `won't fix`

**Comment by MohammedNoureldin** [comment]
@NicolasDorier where? any why?

**Comment by ndeloof** [comment]
`dns` configuration only applies to the default `bridge` network. If you have compose service to use a network (which is the case until you set `network_mode: bridge` the DNS configuration has no impact on container. This is not specific to compose.
Ability to customize DNS resolution would have to apply on the network itself. Might be possible if you set adequate `driver_opts` for this purpose, I haven't checked the `default` driver supports this.

**Comment by MohammedNoureldin** [comment]
Thank you @ndeloof for you response. I checked the `driver_opts` but found no DNS-related options.

https://docs.docker.com/engine/reference/commandline/network_create/
https://docs.docker.com/network/network-tutorial-standalone/

Is there any other undocumented DNS-related options?

**Comment by magec** [comment]
Came across this issue and ended up overwriting the `/etc/resolv.conf` file of the container with a predefined version mounted using a bind, it did the trick.

**Comment by v0ff4k** [comment]
Manually updated(append) `/etc/hosts`  fix my issue, but it works only for couple domain names. Especcially for localhost domain names.
Maybe add to Dockerfile  command to append `/etc/hosts`  solve my problem at all!

**Comment by mvtab** [comment]
Hi, definitely a dirty bug, no matter who says what.

I have the following scenario:
I am doing some tests with `ollama`. Everything with `docker run` worked flawlessly. When trying to port the same setup to docker compose, something wasn't going. After much troubleshooting, I realized the problem was how docker-compose implemented dns. 

## run

I have a custom image where I add my own CA and proxy settings. The following command works flawlessly:
`docker run -d --name test-with-ca -e 'HTTPS_PROXY=http://x.x.x.com:8080' -e 'OLLAMA_HOST=127.0.0.1' -p 11434:11434 ollama-with-ca:0.1`

## compose

I then made the following docker-compose file:
```
name: ai

services:
  ollama:
    image: ollama-with-ca:0.1
    build:
      context: containers
      dockerfile: ollama.Containerfile
    environment:
      HTTPS_PROXY: http://x.x.x.com:8080
      OLLAMA_HOST: 127.0.0.1
    ports: 
    - 127.0.0.1:11434:11434
    pull_policy: never
```
Result: Container has no network connectivity. After troubleshooting a longer time, I discovered the problem was DNS. So I added the following fields to the compose file:
```
    dns:
    - 10.0.6.42
    - 10.0.6.2
    dns_search:
    - x.com
```

Result: still no network connectivity.

## /etc/resolv.conf

After troubleshooting like wild again, I noticed the /etc/resolv.conf files were wrong.
With `docker run` command from above, I get the following (working) /etc/resolv.conf:
```
nameserver 10.0.6.42
nameserver 10.0.6.2
search x.x.com x.lan x.com

# Based on host file: '/etc/resolv.conf' (legacy)
# Overrides: []
```

When running with the docker compose file with DNS settings configured:
```
nameserver 127.0.0.11
search x.com
options ndots:0

# Based on host file: '/etc/resolv.conf' (internal resolver)
# ExtServers: [10.0.6.42 10.0.6.2]
# Overrides: [nameservers search]
# Option ndots from: internal
```

## bridge

Only after I add `network_mode: bridge` to the container, it works as expected and sets my DNS server. 

Before you go ahead and say this is normal, docker compose creates a default network for the compose project: 
```
"Name": "ai_default",
...
"Driver": "bridge"
```

## Conclusion

So compose seems to inherit the right settings. It does not apply them. Setting them explicitly does not work either, until you set the network mode to bridge, even if the default network is a bridge anyway.

**Comment by florihupf** [comment]
I am just here to say me too.  Doing the volume mount workaround `/etc/resolve.conf` worked for me as well.

**Comment by ndeloof** [comment]
Compose only exposes docker engine features, it has no way to disable built-in DNS if engine's API does offer such an option. If you know a way for you to get the expected result with `docker network create` or `docker network connect` command, the same could apply to Compose
